src/full/Agda/Auto/NarrowingSearch.hs

```diff
@@ -453,7 +453,7 @@ topSearch ticks nsol hsol envinfo p searchdepth depthinterval = do
    True -> -- failed immediately
     return False
    False -> do
-    Left solfound <- lift $ searchSubProb [(mainroot, Nothing)] searchdepth
+    Left _ <- lift $ searchSubProb [(mainroot, Nothing)] searchdepth
```

Comment:
> Nice work. One question I had: what is the reasoning for commenting out some imports vs removing others? Shouldn't we just remove all of them? I don't think there is much useful information in what imports there _used to_ be.

The second cleanup is mostly for modules from `base` which can be collected into a `Prelude` module. Modules like `Data.Monoid`, `Data.Traversable` are used for programming in general, and they are fine to stay, I think.

Reply:
> Here it would make sense to write `_solfound` instead of `_`; even though the variable is not used the name serves as documentation.

Good advice. Done.

Reply:
I gave up tracking modules manually. It could be done after decoupling the codebase further.